src/main/java/fr/zcraft/quartzlib/components/nbt/NBTType.java

```diff
@@ -190,17 +243,79 @@ public Object getData(Object nmsNbtTag) {
             return null;
         }
         try {
+            switch (getNmsTagFieldName()) {
+                case "map":
+                    //Since 1.17 "map" became "tags"
+                    //h() return an unmodifiable map
+                    return Reflection.call(nmsNbtTag.getClass(), nmsNbtTag, "h");
```

Comment:
We can only get an unmodifiable map instead of the actual map, so I had to do by hand the insertion in the map using methods like setInt .